biocypher_metta/adapters/hgnc_processor.py

```diff
@@ -2,93 +2,125 @@
 import csv
 import pickle
 import os
-from typing import Dict, Optional, Any
+from typing import Dict, Any
+from io import StringIO
+from datetime import datetime, timedelta
 
 class HGNCSymbolProcessor:
     def __init__(self, pickle_file_path: str = 'hgnc_gene_data/hgnc_data.pkl', version_file_path: str = 'hgnc_gene_data/hgnc_version.txt'):
         self.pickle_file_path = pickle_file_path
         self.version_file_path = version_file_path
-        self.current_symbols: Dict[str, str] = {}  # Current official symbols
-        self.symbol_aliases: Dict[str, str] = {}   # Maps old symbols to current ones
-        self.ensembl_to_symbol: Dict[str, str] = {}  # Maps Ensembl IDs to HGNC symbols
-        self.current_version = self.get_current_version()
-
-    def get_current_version(self) -> str:
-        """Get the current version from HGNC server"""
-        url = "https://ftp.ebi.ac.uk/pub/databases/genenames/hgnc/json/hgnc_complete_set.json"
-        response = requests.head(url)
-        return response.headers.get('Last-Modified', '')
-
-    def check_version(self) -> bool:
-        """Check if we need to update the data"""
+        self.current_symbols: Dict[str, str] = {}
+        self.symbol_aliases: Dict[str, str] = {}
+        self.ensembl_to_symbol: Dict[str, str] = {}
+        self.update_interval = timedelta(days=7)  # Update weekly
+        self.last_update_check = None
+        self.last_check_result = None
+
+    def check_update_needed(self) -> bool:
+        """Check if we need to update the data based on the last update time"""
+        current_time = datetime.now()
+        
+        # Check if we have a cached result from the last 30 minutes
+        if self.last_update_check and (current_time - self.last_update_check) < timedelta(minutes=30):
```

Comment:
Could you increase the needed time to update the cache. For large datasets it might take hours if not days so we don't need it to update in the middle of running among different adapters. I suggest 48 hours would be okay.